common/searchattribute/mapper.go

```diff
@@ -39,11 +40,58 @@ type (
 		GetAlias(fieldName string, namespace string) (string, error)
 		GetFieldName(alias string, namespace string) (string, error)
 	}
+
+	identityMapper struct{}
+
+	MapperProvider interface {
+		GetMapper(namespaceName string) (Mapper, error)
```

Comment:
```suggestion
		GetMapper(nsName namespace.Name) (Mapper, error)
```
It shouldn't cause circular dependency now. `Mapper` uses `string` because it is externally exposed interface. All internal func should pass namespace name and Id using dedicated type.